src/unix/thread.c

```diff
@@ -387,16 +387,19 @@ int uv_sem_init(uv_sem_t* sem, unsigned int value) {
   uv_sem_t semid;
   struct sembuf buf;
   int err;
+  union {
+    int val;
+    struct semid_ds *buf;
+    unsigned short *array;
```

Comment:
Style: star leans left.  I'm curious though, is there a reason you can't use `union semun arg;` here?